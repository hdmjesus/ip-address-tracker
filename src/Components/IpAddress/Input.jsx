import React from "react";
import iconArrrow from "../../Statics/images/icon-arrow.svg";
export const IpAddressInput = () => {
  return (
    <form className="  m-auto flex w-full ">
      <div className="flex min-w-200 justify-between bg-white rounded-lg  h-9 m-auto  w-full   ">
        <input
          className=" w-full p-2 text-xs bg-transparent outline-none "
          type="text"
          placeholder="Search for Any IP Address or Domain"
        />
        <button type="submit" className="bg-black h-9 w-9 rounded-r-lg">
          <img src={iconArrrow} alt="iconArrow" className="w-3 m-auto" />
        </button>
      </div>

      {/* <button type="submit" className="bg-black h-9 w-9">
        <img src={iconArrrow} alt="iconArrow" className="w-3" />
      </button> */}
    </form>
  );
};
