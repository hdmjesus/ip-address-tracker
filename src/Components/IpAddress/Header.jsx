import React from "react";
import { IpAddressInput } from "./Input";
import { IpAddressResultsContainer } from "./Results/Container";
import bgHeader from "../../Statics/images/pattern-bg.png";

export const IpAddressHeader = () => {
  return (
    <header
      className="w-full h-80 p-7  m-auto z-10"
      style={{
        backgroundImage: `url(${bgHeader})`,
        backgroundRepeat: "no-repeat",
      }}
    >
      <h1 className="  font-rubik font-bold text-white text-xl text-center mb-4">
        IP Address Tracker
      </h1>
      <IpAddressInput />
      <IpAddressResultsContainer />
    </header>
  );
};
