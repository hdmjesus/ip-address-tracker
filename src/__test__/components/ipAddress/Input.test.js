import React from "react";
import { mount } from "enzyme";
import { IpAddressInput } from "../../../Components/IpAddress/Input";
import { TextContext } from "../../../Context/StaticContext";

describe("Pruebas en <IpAddressInput/>", () => {
  const setValueInput = jest.fn();

  const { Provider } = TextContext;
  let loading = false;
  let data = null;
  const wrapper = mount(
    <Provider value={{ data, loading, setValueInput }}>
      <IpAddressInput />
    </Provider>
  );

  test("should show good the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("dont should call handleSubmit", () => {
    const formSubmit = wrapper.find("form").prop("onSubmit");
    expect(setValueInput).toHaveBeenCalledTimes(0);
  });

  test("should call handleSubmit", () => {
    const formSubmit = wrapper.find("form").prop("onSubmit");
    formSubmit({ preventDefault() {} });
    expect(setValueInput).toHaveBeenCalledTimes(1);
  });

  test("should call reset function input", () => {
    expect(wrapper.find("input").prop("value")).toBe("");
  });
  test("should cambiar el estado interno del input correctamente", () => {
    const value = "Google.com";

    let input = wrapper.find("input").simulate("change", {
      target: { value: value, name: "ipInput" },
    });
    expect(wrapper.find("input").prop("value")).toBe(value);
    // expect(wrapper.find('input').prop('value')).toBe('');
  });
});
