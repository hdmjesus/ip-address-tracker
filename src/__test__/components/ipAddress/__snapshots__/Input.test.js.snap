// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <IpAddressInput/> should show good the component 1`] = `
<form
  className="  m-auto flex w-full "
  onSubmit={[Function]}
>
  <div
    className="flex min-w-200 justify-between bg-white rounded-lg  h-9 m-auto  w-full   "
  >
    <input
      className=" w-full p-2 text-xs bg-transparent outline-none "
      name="ipInput"
      onChange={[Function]}
      placeholder="Search for Any IP Address or Domain"
      type="text"
      value=""
    />
    <button
      className="bg-black h-9 w-9 rounded-r-lg"
      type="submit"
    >
      <img
        alt="iconArrow"
        className="w-3 m-auto"
        src="icon-arrow.svg"
      />
    </button>
  </div>
</form>
`;
